exporter/exporter.go

```diff
@@ -276,6 +276,10 @@ func (b *Builder) CreateLogs(ctx context.Context, set CreateSettings) (Logs, err
 	return f.CreateLogsExporter(ctx, set, cfg)
 }
 
+func (b *Builder) Config(componentID component.ID) component.Config {
```

Comment:
Independent of the config solution we discuss, I am curious why you need this:
1. Can you just check if a factory for the `componentID.Type()` is present?
2. I think you don't need access to the config. Should you then just have a `IsConfigured(componentId) bool` if the first solution is not enough?

Reply:
I explained in more detail [here](https://github.com/open-telemetry/opentelemetry-collector/pull/6927#issuecomment-1380799272), but the minimal need here is to identify the connectors in the `receivers` and `exporters` sections of each pipeline. 

1. This would work if a we had a guarantee that connector types cannot overlap with receiver or exporter types. This is possibly a reasonable rule, but I think we'd have to enforce this during factory registration. Not sure we care but I think this would be incompatible with an asynchronous/dynamic factory registration pattern. 
2. `IsConfigured` is enough. In fact, I only need it for connectors specifically. I can switch to that - I just thought it was oddly specific and that in the end supporting `Config` alongside `Factory` was natural.

I don't feel strongly about the implementation, so either of these approaches is fine with me.

Reply:
Ok, I've reworked #6700 to prove to myself that `connector.Builder.IsConfigured` is the only thing absolutely necessary here, so I'm going to close this PR and include that in another PR that uses it.